fw/http_frame.c

```diff
@@ -2013,22 +1992,86 @@ tfw_h2_insert_frame_header(struct sock *sk, TfwH2Ctx *ctx, TfwStream *stream,
 	return r;
 }
 
+static inline unsigned int
+tfw_h2_calc_min_to_send(struct sock *sk, TfwH2Ctx *ctx)
+{
+	/* Empirically chosen value. */
+	const unsigned int min_to_send_dflt = 512;
+	struct dst_entry *dst = __sk_dst_get(sk);
+	unsigned int min_to_send;
+
+	/* MTU is unknown, we should send any available bytes. */
+	if (unlikely(!dst))
+		return 0;
+
+	/*
+	 * Tempesta FW avoid to send frame if the size of frame
+	 * is less then 512 bytes, except when mtu is small or http2
+	 * initial window is less then 1024 bytes (client usually sends
+	 * window update frame with a size equal to a half of initial
+	 * window).
+	 */
+	min_to_send = min3(min_to_send_dflt, dst_mtu(dst) >> 4,
```

Comment:
You doing `dst_mtu()`, but maybe it would be better to pass `mss_now` to this function to respect tcp headers size, etc?